src/main/java/org/apache/commons/imaging/ImagingParameters.java

```diff
@@ -0,0 +1,333 @@
+     * @return Valid values: Any format defined in ImageFormat, such as
+     * ImageFormat.IMAGE_FORMAT_PNG.
+     */
+    public ImageFormat getImageFormat() {
+        ImageFormat value = this.imageFormat;
+        checkIfParameterIsPresent(value);
+        return value;
+    }
+    
+    /**
+     * Parameter used in write operations to indicate desired image format.
+     * @param value Valid values: Any format defined in ImageFormat, such as
+     * ImageFormat.IMAGE_FORMAT_PNG.
+     */
+    public void setImageFormat(final ImageFormat value) {
+        checkIfValueIsNull(value);
+        this.imageFormat = value;
+    }
+    
+    //****** bufferedImageFactory ******
+    
+    /**
+     * 
+     * @return {@code true} if there is a value present, {@false} else.
+     */
+    public boolean isBufferedImageFactoryPresent() {
+        return this.bufferedImageFactory != null;
+    }
+    
+    /**
+     * 
+     * @return A BufferedImageFactory
+     */
+    public BufferedImageFactory getBufferedImageFactory() {
+        BufferedImageFactory value = this.bufferedImageFactory;
+        checkIfParameterIsPresent(value);
+        return value;
+    }
+    
+    /**
+     * 
+     * @param value A BufferedImageFactory
+     */
+    public void setBufferedImageFactory(final BufferedImageFactory value) {
+        checkIfValueIsNull(value);
+        this.bufferedImageFactory = value;
+    }
+    
+    //****** pixelDensity ******
+    
+    /**
+     * Parameter key. Used in write operations to indicate the desired pixel
+     * density (DPI), and/or aspect ratio.
+     * @return {@code true} if there is a value present, {@false} else.
+     * @see org.apache.commons.imaging.PixelDensity
+     */
+    public boolean isPixelDensityPresent() {
+        return this.pixelDensity != null;
+    }
+    
+    /**
+     * Parameter key. Used in write operations to indicate the desired pixel
+     * density (DPI), and/or aspect ratio.
+     * @return Valid values: PixelDensity
+     * @see org.apache.commons.imaging.PixelDensity
+     */
+    public PixelDensity getPixelDensity() {
+        PixelDensity value = this.pixelDensity;
+        checkIfParameterIsPresent(value);
+        return value;
+    }
+    
+    /**
+     * Parameter key. Used in write operations to indicate the desired pixel
+     * density (DPI), and/or aspect ratio.
+     * @param value Valid values: PixelDensity
+     * @see org.apache.commons.imaging.PixelDensity
+     */
+    public void setPixelDensity(final PixelDensity value) {
+        checkIfValueIsNull(value);
+        this.pixelDensity = value;
+    }
+    
+    //****** check methods ******
+    
+    /**
+     * Throws a RuntimeException if a given value is {code null}.
+     * @param value 
+     */
+    final void checkIfValueIsNull(final Object value) {
+        if (value == null) {
+            throw new IllegalArgumentException("The value for any parameter must not null.");
```

Comment:
On 23.02.2015 20:00, Benedikt Ritter
      wrote:
    
    
      In src/main/java/org/apache/commons/imaging/ImagingParameters.java:
      > +     * @see org.apache.commons.imaging.PixelDensity
> +     */
> +    public void setPixelDensity(final PixelDensity value) {
> +        checkIfValueIsNull(value);
> +        this.pixelDensity = value;
> +    }
> +    
> +    //****** check methods ******
> +    
> +    /**
> +     * Throws a RuntimeException if a given value is {code null}.
> +     * @param value 
> +     */
> +    final void checkIfValueIsNull(final Object value) {
> +        if (value == null) {
> +            throw new IllegalArgumentException("The value for any parameter must not null.");

      Would be helpful for users to know which parameter caused the
        exception.
      —
        Reply to this email directly or view
          it on GitHub.
      
        
          
          
        
        
      
    
    Added it :-)
    
    -- 
Michael Groß